command-parser.js

```diff
@@ -113,6 +113,22 @@ class CommandContext {
 		this.inputUsername = '';
 	}
 
+	checkFormat(room, message) {
+		if (!room) return false;
+		if (!room.stretching && !room.capitals) return false;
+		let formatError = false;
+		let stretchMatch = room.stretching && message.match(/(.+?)\1{7,}/i);
+		let capsMatch = room.capitals && message.match(/[A-Z\s]{18,}/);
```

Comment:
7 is from stretching which is used by most bots.  I believe old TTT bot used to use 18 for caps, but changed it to 12 more recently.   Another possible mark would be 19, to avoid filtering out names that are 19 chars in all caps

For the messages you should do this thing from TTT's bot:
``msg = msg.trim().replace(/[ \u0000\u200B-\u200F]+/g, ' '); // removes extra spaces and null characters so messages that should trigger stretching do so``
where it removes any null characters before doing the stretching check (ex. kota's "ok" has many extra null characters at the end iirc)